### DISCUSSION TITLE
PyTorch is deprecating the `pytorch` conda channel

### DISCUSSION THREAD

**Discussion by roaldarbol** [mooch443/trex#229] [General]
Hi @mooch443!
Saw a couple of days ago that PyTorch is sunsetting their PyTorch conda channel - and then today I remembered that you rely on that channel (at least last I checked). So yeah, just a heads up in case you hadn't heard it yet: https://github.com/pytorch/pytorch/issues/138506

They recommend transferring to the community maintained conda-forge channel instead (they're also helping out during that transition it seems).

All the best, 
Mikkel

**Comment by mooch443** [comment]
> Hi @mooch443!
> 
> Saw a couple of days ago that PyTorch is sunsetting their PyTorch conda channel - and then today I remembered that you rely on that channel (at least last I checked). So yeah, just a heads up in case you hadn't heard it yet: https://github.com/pytorch/pytorch/issues/138506
> 
> 
> 
> They recommend transferring to the community maintained conda-forge channel instead (they're also helping out during that transition it seems).
> 
> 
> 
> All the best, 
> 
> Mikkel

Hey! Thanks for the info, I didn't know that. But: I already moved away from using it. There was no point anymore, since it makes dependencies quite complicated and just increases the workload for updating when I can also simply use pip. I hope that won't be deprecated :)

We transitioned entirely to conda-forge for other reasons: especially license issues. It seems you may be liable to pay if you use conda defaults/main in your work environment. So yea. Stay tuned for updates regarding this - I am writing the documentation as we speak. Final steps.

Thanks for your message anyway, hope you're doing good!

-Tristan

**Reply by roaldarbol** [reply]
Ah, perfect, good to hear! It seems they will recommend using pip as their main officially supported version. But I'd still love for it to be on conda-forge, and wouldn't mind helping to package it / maintain the recipe once I've submitted my thesis in January.

**Reply by mooch443** [reply]
you mean to have pytorch on conda-forge? particularly *including* a gpu version would be nice.

**Reply by roaldarbol** [reply]
Actually meant trex 😅 I’m however also following the pytlrch conda-forge situation closely now, and the main barrier now seems to be getting Windows builds to work.

**Reply by roaldarbol** [reply]
pytorch currently works for both gpu and cpu on Linux and Metal MacOS.

**Reply by mooch443** [reply]
> Actually meant trex 😅 I’m however also following the pytlrch conda-forge situation closely now, and the main barrier now seems to be getting Windows builds to work.

Oh TRex, yes you are welcome to help with that of course. I know how to make a conda package but I don't have much of a clue regarding conda forge. Let me know what you need to get this to work if you want to spend time on it :) 

With the PyTorch situation, it could be beneficial also for TRex to have it install via conda - it's not a huge issue right now though. Is there an issue for it I can follow? I think I found one some time ago.

**Reply by mooch443** [reply]
> pytorch currently works for both gpu and cpu on Linux and Metal MacOS.

If installed via pip, yes. Do you mean forge? The normal conda versions (PyTorch channel) have licensing issues since not being hosted on forge. So these are out of the question. I am happy with what I have now, since it seems to work on all major OSes I have tested.

**Reply by roaldarbol** [reply]
> > pytorch currently works for both gpu and cpu on Linux and Metal MacOS.
> 
> If installed via pip, yes. Do you mean forge? The normal conda versions (PyTorch channel) have licensing issues since not being hosted on forge. So these are out of the question. I am happy with what I have now, since it seems to work on all major OSes I have tested.

The actual conda-forge version of pytorch. The relevant issue (Windows builds on conda-forge) is this: https://github.com/conda-forge/pytorch-cpu-feedstock/issues/32. The primary PR working on it is here: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/2311

I'm spending a bit of time learning conda-forge recipe maintenance, so am by no means an expert, but I'm learning and will be happy to get it up and running (in Jan/Feb). It's also simply for convenience of end-users "shopping" around trying different tools - currently they're installed in many different ways, in incompatible envs etc. I'd really like being able to do e.g. `pixi global install trex`. I've gotten that working for idtrackerai, although without Windows builds, and deeplabcut (I'll document it soonish).

**Reply by mooch443** [reply]
> The actual conda-forge version of pytorch. The relevant issue (Windows builds on conda-forge) is this: https://github.com/conda-forge/pytorch-cpu-feedstock/issues/32. The primary PR working on it is here: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/2311
> 

Thanks!

> I'm spending a bit of time learning conda-forge recipe maintenance, so am by no means an expert, but I'm learning and will be happy to get it up and running (in Jan/Feb). It's also simply for convenience of end-users "shopping" around trying different tools - currently they're installed in many different ways, in incompatible envs etc. I'd really like being able to do e.g. `pixi global install trex`. I've gotten that working for idtrackerai, although without Windows builds, and deeplabcut (I'll document it soonish).

I mean seeing as TRex successfully builds on all OSes using simple `conda build .`, it should be possible to transfer to any respectable build system. It just needs appropriate C++/C compilers, CMake (unfortunately), some OpenGL and Python. As well as their dependencies I guess.

Happy to work on this in January too. First comes the release on normal conda + priority on maybe writing about it elsewhere :)
